stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/FootnoteView.kt

```diff
@@ -67,6 +69,52 @@ internal class FootnoteView : ConstraintLayout {
             threadRepliesButton.text =
                 resources.getQuantityString(R.plurals.stream_ui_thread_messages_indicator, replyCount, replyCount)
         }
+        setupUserAvatars(isMine, threadParticipants)
+    }
+
+    private fun setupUserAvatars(isMine: Boolean, threadParticipants: List<User>) {
+        fun applyUser(user: User?, avatarView: AvatarView) {
+            if (user != null) {
+                avatarView.setUserData(user)
+            } else {
+                avatarView.isVisible = false
+            }
+        }
+
+        with(threadsFootnote) {
+
+            root.updateConstraints {
+                clear(threadRepliesButton.id, ConstraintSet.TOP)
+                connect(
+                    threadRepliesButton.id,
+                    ConstraintSet.TOP,
+                    if (isMine) firstMineUserImage.id else firstTheirUserImage.id,
+                    ConstraintSet.TOP
+                )
+            }
+
+            firstTheirUserImage.isVisible = !isMine
+            secondTheirUserImage.isVisible = !isMine
+            firstMineUserImage.isVisible = isMine
+            secondMineUserImage.isVisible = isMine
+
+            val (first, second) = getTwoLastUsers(threadParticipants)
+
+            applyUser(first, if (isMine) firstMineUserImage else firstTheirUserImage)
+            applyUser(second, if (isMine) secondMineUserImage else secondTheirUserImage)
+        }
+    }
+
+    private fun getTwoLastUsers(threadParticipants: List<User>): Pair<User?, User?> {
```

Comment:
Not sure where things go wrong exactly with the data, but when there's a single reply to a thread, we end up showing two avatars, one of them being the sender of the original message:

![image](https://user-images.githubusercontent.com/12054216/104822230-46455700-5841-11eb-8723-9b8efc428d71.png)

Reply:
I think it was done intentionally. So there is description by designers https://www.loom.com/share/9cadf788a5db4d0192b946dd9822d409
As I didn't do anything with data propagated from BE seems it is an implementation of requirements.

Reply:
Ah, okay, I think this is confusing as a user (looks like more than 1 reply), but apparently it's intentional then